File: FitBook.Model/Requests/TrainingEquipment/TrainingEquipmentInsertRequest.cs
namespace FitBook.Model.Requests.TrainingEquipment;

public class TrainingEquipmentInsertRequest
{
    public bool IsRequired { get; set; }
    public string? Note { get; set; }
    public int TrainingId { get; set; }
    public int EquipmentId { get; set; }
}

File: UI/fitbook_mobile/lib/models/requests/training_equipment_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

import '../common/api_request_body.dart';

part 'training_equipment_insert_request.g.dart';

@JsonSerializable()
class TrainingEquipmentInsertRequest implements ApiRequestBody {
  TrainingEquipmentInsertRequest({
    required this.isRequired,
    this.note,
    required this.trainingId,
    required this.equipmentId,
  });

  final bool isRequired;
  final String? note;
  final int trainingId;
  final int equipmentId;

  factory TrainingEquipmentInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$TrainingEquipmentInsertRequestFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$TrainingEquipmentInsertRequestToJson(this);
}
